lib/utils/cache.js

```diff
@@ -0,0 +1,139 @@
+'use strict';
+const LRUMap = require('lru-cache');
```

Comment:
Redis is an:

> in-memory data structure store

The plan is to still use Redis (although I think it would be slower than this solution for the London case - more efficient when we introduce fail over and resiliency), this is just part 1.  

Because this has some protections built into it, it will reduce the need to go to redis on every request if we receive many requests for the same URL at the same time.  Without this, if we get a huge burst of traffic, all that load would go on redis - here, if we receive 1000 requests in one instant, we'll only make one backend request and we won't need to consult redis 1000 times.